synapse/cortex.py

```diff
@@ -920,6 +920,22 @@ async def initServiceStorage(self):
         self.pkghive = await pkghive.dict()
         self.svchive = await svchive.dict()
 
+        self.deprlocks = await self.hive.get(('cortex', 'model', 'deprlocks'), {})
+        # TODO: 3.0.0 conversion will truncate this hive key
+        for name, locked in self.deprlocks.items():
+
+            form = self.model.form(name)
+            if form is not None:
+                form.locked = locked
+
+            prop = self.model.prop(name)
+            if prop is not None:
+                prop.locked = locked
+
+            _type = self.model.type(name)
```

Comment:
its possible to have a form deprecated and the type itself not be deprecated. this name reuse would lock *both* of them.